component/azstorage/file_share.go

```diff
@@ -266,6 +294,36 @@ func (fs *FileShare) DeleteDirectory(name string) (err error) {
 	log.Trace("FileShare::DeleteDirectory : name %s", name)
 
 	dirURL := fs.Share.NewDirectoryURL(filepath.Join(fs.Config.prefixPath, name))
+
```

Comment:
Does DeleteDirectory work recursively? Does it give you an error if the directory is not empty?

Reply:
Yes, DeleteDirectory works recursively. No, it doesn't give an error if the directory is not empty because for any directory it deletes, it'll delete all the files and recursively delete any subdirectories.

Reply:
I think I meant if the file share SDK API works recursively. I'm wondering if we can replace all this with one call to the SDK basically

Reply:
Oh gotcha. Yes it does! I deleted all the recursive stuff and just called the SDK in the most recent commits

Reply:
*Sorry got this mixed up with RenameDirectory(). No, the SDK doesn't work recursively; the directory must be empty before being deleted by the SDK call.